parquet-hadoop/src/main/java/org/apache/parquet/hadoop/InternalParquetRecordReader.java

```diff
@@ -265,4 +273,51 @@ public boolean nextKeyValue() throws IOException, InterruptedException {
     return Collections.unmodifiableMap(setMultiMap);
   }
 
+  /**
+   * Returns the ROW_INDEX of the current row.
+   */
+  public long getCurrentRowIndex() {
+    if (current == 0L) {
```

Comment:
What is the reason not turning -1?

Reply:
current is an existing variable which tracks number of rows already processed. It is initialized to 0 at declaration time. So here we are trying to see if it is still 0, that means we haven't processed any row yet.

Reply:
I understand why we are checking 'current == 0L'. I was asking why you choose throw exception other than returning an invalid value. This is a public method. We should have it documented ether way you choose.

Reply:
There was no specific reason for choosing exception over -1.
I have updated it to return -1 and also updated all the public method docs to reflect the same.